### DISCUSSION TITLE
Example Typescript project

### DISCUSSION THREAD

**Discussion by rg-wood** [ing-bank/lion#2318] [General] [answered]
I'm having some issues getting Lion to play nice with Typescript. Does anyone have any example Typescript projects I could have a look at?

**Comment by tlouisse** [comment]
Hi, we don't have an example project, but I can share the tsconfig.json of our own extension layer if that helps you:

```json
{
  "compilerOptions": {
    "target": "ESNext",
    "module": "ESNext",
    "moduleResolution": "NodeNext",
    "lib": ["es2017", "dom"],
    "allowJs": true,
    "checkJs": true,
    "strict": true,
    "noImplicitThis": true,
    "composite": true,
    "emitDeclarationOnly": true,
    "alwaysStrict": true,
    "types": ["node", "mocha", "chai", "sinon"],
    "esModuleInterop": true,
    "suppressImplicitAnyIndexErrors": true,
    "skipLibCheck": true,
    "incremental": true,
    "declaration": true,
    "declarationMap": true,
    "sourceMap": true
  },
  "exclude": ["node_modules"]
}
```

**Reply by rg-wood** [reply]
Thanks for the response. I'll take a look.

What is the correct way to import lion components? For example, this is what I'm trying:

```js
import { LionDrawer } from '@lion/ui'
```

I have a simple failing test case here:

https://github.com/rg-wood/lion-issue-reproduction-fork-lyygsvnq/blob/main/test.ts

**Reply by tlouisse** [reply, marked as answer]
Not sure if it solves your issue, but you forgot the entrpoint in your example. Can you try:

```js
import { LionDrawer } from '@lion/ui/drawer.js';

console.log(LionDrawer);
```

**Reply by rg-wood** [reply]
Ah, thank you, that worked with `moduleResolution` set to `nodenext`.
